### ISSUE TITLE
Support to download file from GitHub releases

### ISSUE THREAD

**Issue by LinuxSuRen** [jsdelivr/jsdelivr#18203]
It's very very slow for some areas, for example: the users from China. So, I was wondering if it's possible to allow users download files from GitHub releases by `jsdelivr`.

**Comment by MartinKolarik** [comment]
If you mean the additionally uploaded release assets that isn't possible right now. But anything that's stored in the repo is available.

**Comment by LinuxSuRen** [comment]
Yes, we can download the release assets directly. It just very slow for us.

**Comment by MartinKolarik** [comment]
You can always put them to the repo, to a separate branch. E.g., some projects have a `release` branch where they commit all release files and then those can be accessed via the CDN.

**Comment by LinuxSuRen** [comment]
Yes, that's a good solution for now. Actually I've done that in my repo. But generally, we need to download from many other different repos. Anyway, this is just a suggestion.

**Comment by MartinKolarik** [comment]
I'll keep this open as a possible enhancement but can't say yet if or when it'll be added (likely not in the near future).

**Comment by LinuxSuRen** [comment]
Thanks for taking my suggestion. It will be very convenience if `jsdelivr` could support GitHub release asset. I totally understand that adding such a feature might be too fast.
Further more, I resolved my problems by adding a Nexus raw proxy repo. Just in case someone else might need a similar solution, I put the screenshot of the Nexus setting below:
![image](https://user-images.githubusercontent.com/1450685/78847629-f0d25300-7a41-11ea-87b5-e95b77457503.png)

**Comment by foisonocean** [comment]
vote for this

**Comment by RodrigoDornelles** [comment]
It would be very useful also for projects written in several languages that are compiled for webassembly and published in releases.

**Comment by zh99998** [comment]
+1 for this request.

**Comment by axetroy** [comment]
Two years have passed, it seems that JSDELIVR steam doesn't want to do this. I have to find another way

**Comment by RodrigoDornelles** [comment]
> Two years have passed, it seems that JSDELIVR steam doesn't want to do this. I have to find another way

Hi @axetroy 

am solved this by publishing every release in npm, and then pulling via JSDELIVR

<https://github.com/RodrigoDornelles/3bc-lang/blob/master/.github/workflows/automatic-publish.yaml>

**Comment by Pandapip1** [comment]
+1, this is still wanted.

**Comment by Xyueta** [comment]
Any updates? It would be necessary in our project, when we have to use the alternative for GitHub for the release, especially for users in China region.

**Comment by scripthunter7** [comment]
+1, I also support this feature request, because by default if you release a `.txt` or `.js` file on GitHub as a release asset, after that it is not so easy to download that file, since you cannot open it as "raw" in the browser, since the browser always forces the "Download prompt" , as a redirect happens in the background. I also asked this question in the GitHub community, but I didn't get an answer: https://github.com/orgs/community/discussions/37100
If jsDelivr would support GH releases in the future, this problem could be solved in a much more convenient way.
It would probably make sense, since as several people mentioned before my comment, many build results are uploaded as release assets.

**Comment by UlyssesZh** [comment]
+1 for this request. My reason is that the responds from GitHub releases assets do not have the necessary headers for CORS access, so we cannot directly use the URL to GitHub reseases assets.

**Comment by hiddentao** [comment]
+1 would be great to have this

**Comment by codeliger** [comment]
why is this still open?

**Comment by LinuxSuRen** [comment]
Five years passed

**Comment by Pandapip1** [comment]
If you sort all the issues (open and closed) by number of 👍 reactions, this is the second most :+1:'d issue other than an actual SSL issue that caused jsdelivr to stop working entirely:

https://github.com/jsdelivr/jsdelivr/issues?q=is%3Aissue%20sort%3Areactions-%2B1-desc

Any chance the priority on this issue could be increased to high?

**Comment by jimaek** [comment]
Guys please note that this is not as simple as adding a new proxy endpoint and supporting GitHub releases. There are many reasons why the implementation of this feature is not as simple and has been a lower priority:
- Github rate-limits. We already spend a lot of resources trying to balance our system and avoid hitting GitHub's rate-limits. They are extermely low for our use-cases. Releases would result in additional API calls that could create serious issues for all existing Github endpoints.
- Scale. At our scale we cant just build something that works for 100 or 10k users, it needs to be able to scale to billions of requests without problems. Combine that with the rate-limits issue and you can see how that complicates things.
- Alternatives. It is very easy to use jsDelivr with GitHub without Releases. Just commit the files to a `dist/` folder or create a `release` branch and use it through jsDelivr. Or even better push to NPM, it is a lot easier for us to download files and cache them from NPM than dealing with GitHub's API to resolve versions, tags, etc... 

I hope you understand why this might sound like a simple addition but in reality is a complex and problematic topic.

**Comment by VulcanoSoftware** [comment]
still not this fundamental feature in the big 26?

**Comment by Pandapip1** [comment]
> I hope you understand why this might sound like a simple addition but in reality is a complex and problematic topic.

I recognize that this isn't a simple issue and obviously none of us have the right to force you guys do do any specific work, but the demand is here; please consider at least updating the labels.

FWIW, while I haven't interacted with them directly, several of my peers have and have reported that GitHub is generally pretty responsive and accommodating of increasing rate limits where needed. Otherwise, the other choice is to have repository admins install a GitHub app. Each installation (read: user account or org) has its own rate limit bucket.

> Just commit the files to a dist/ folder or create a release branch

-1; committing build artifacts to source control [is bad practice](https://gitbybit.com/gitopedia/best-practices/know-what-to-ignore) and should not be encouraged
